src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -576,9 +590,7 @@ public static JsonTypeInfo CreateJsonTypeInfo(Type type, JsonSerializerOptions o
                 ThrowHelper.ThrowArgumentException_CannotSerializeInvalidType(nameof(type), type, null, null);
             }
 
-            s_createJsonTypeInfo ??= typeof(JsonTypeInfo).GetMethod(nameof(CreateJsonTypeInfo), new Type[] { typeof(JsonSerializerOptions) })!;
-            return (JsonTypeInfo)s_createJsonTypeInfo.MakeGenericMethod(type)
```

Comment:
Just a note that this PR is changing the refactoring done in https://github.com/dotnet/runtime/pull/67526.

In effect, replacing `MethodInfo.MakeGenericMethod` with `Activor.CreateInstance()`?

Reply:
It's using `Type.MakeGenericType` + `Activator` instead of `MethodInfo.MakeGenericMethod`, but everything is behind logic marked RUC so it shouldn't matter much.